src/evalfunc.c

```diff
@@ -5507,12 +5510,25 @@ f_getregion(typval_T *argvars, typval_T *rettv)
 	    || check_for_opt_dict_arg(argvars, 2) == FAIL)
 	return;
 
-    if (list2fpos(&argvars[0], &p1, &fnum, NULL, FALSE) != OK
-	    || (fnum >= 0 && fnum != curbuf->b_fnum))
+    if (list2fpos(&argvars[0], &p1, &fnum1, NULL, FALSE) != OK)
 	return;
 
-    if (list2fpos(&argvars[1], &p2, &fnum, NULL, FALSE) != OK
-	    || (fnum >= 0 && fnum != curbuf->b_fnum))
+    if (list2fpos(&argvars[1], &p2, &fnum2, NULL, FALSE) != OK)
+	return;
+
+    if (fnum1 != fnum2)
+	return;
+
+    buf = buflist_findnr(fnum1);
+    FOR_ALL_TABPAGES(tp)
+    {
+        FOR_ALL_WINDOWS(wp)
+        {
+            if (wp->w_buffer == buf)
+                break;
+        }
+    }
+    if (wp == NULL)
```

Comment:
It may be necessary to use current window if the specified bufnum is 0, as if the loop finds a non-current window, it's 'list' or 'virtualedit' options may be different.

Reply:
Fixed.